include/curl/typecheck-gcc.h

```diff
@@ -367,8 +377,16 @@ _CURL_WARNING(_curl_easy_getinfo_err_curl_socket,
   (CURLINFO_DOUBLE < (info) && (info) < CURLINFO_SLIST)
 
 /* true if info expects a pointer to struct curl_slist * argument */
-#define _curl_is_slist_info(info)                                             \
-  (CURLINFO_SLIST < (info) && (info) < CURLINFO_SOCKET)
+#define _curl_is_slist_info(info)                                       \
+  (CURLINFO_SLIST < (info) && (info) < CURLINFO_SOCKET &&               \
+   ((info & CURLINFO_MASK) < (CURLINFO_CERTINFO & CURLINFO_MASK)))
```

Comment:
I find _curl_is_slist_info a bit complicated now.

IIUC the problem here is that three infos with type CURLINFO_SLIST got misused for other things and that cannot be changed for backwards compatibility?

Then maybe just these three infos could be excluded here instead of excluding all future infos? Then CURLINFO_SLIST can be used for curl_slist pointers again and future other non-slist options can use something new.

Or make CURLINFO_SLIST equivalent to CURLOPTTYPE_OBJECTPOINT if it's just a pointer to anything? Then this could maybe just be == CURLINFO_SSL_ENGINES || == CURLINFO_COOKIELIST, following the style for the CURLOPTTYPE_OBJECTPOINT options.